pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/plan/PlanRequestContext.java

```diff
@@ -87,4 +87,8 @@ public void addReceivingMailboxes(List<MailboxIdentifier> ids) {
   public List<MailboxIdentifier> getReceivingMailboxes() {
     return ImmutableList.copyOf(_receivingMailboxes);
   }
+
+  public OperatorExecutionContext getOperatorExecutionContext() {
+    return new OperatorExecutionContext(this);
+  }
```

Comment:
do not lazy generate this. create the operator execution context at construct time.